fdbserver/workloads/FuzzApiCorrectness.actor.cpp

```diff
@@ -367,15 +376,26 @@ struct FuzzApiCorrectnessWorkload : TestWorkload {
 								//TraceEvent("WDRInitBatch").detail("I", i).detail("CommittedVersion", tr->getCommittedVersion());
 								break;
 							} catch (Error& e) {
+								if (e.code() == error_code_illegal_tenant_access) {
+									ASSERT(!self->writeSystemKeys);
```

Comment:
Is the idea here that we will get an illegal access if we set `useSystemKeys` (because we will enable raw access transactions) but we don't actually set the prefix in the system keys (i.e. `writeSystemKeys` is false)? If so, can this be extended to say:

```suggestion
									ASSERT(!self->writeSystemKeys && self->useSystemKeys);
```

Reply:
Even though `useSystemKeys==false`, if the tenant -1 still can set the option `raw_access` and writes data to normal keyspace and trigger `illegal_tenant_access`. The value of `useSystemKeys` doesn't matter, the truth is if tenant -1 doesn't write to system keyspace `writeSystemKeyspace=false` and get the raw access, it will come to this block.

Reply:
If I understand correctly, the tenant `-1` will always be a raw transaction, but it should only have `raw_access` if `useSystemKeys == true` (at least in this loop) as a result of setting the `ACCESS_SYSTEM_KEYS` option. Though to be honest, I'm not actually sure why that would work, since if the tenant mode is required it should cause this to fail.

If that is the case, then we should only be able to get `illegal_tenant_access` if the transaction was granted `raw_access`, which would require `useSystemKeys == true`.

Reply:
I gather that since you changed it and changed it back, it caused a failure; do you know where raw access is being enabled?

Reply:
I don't think that would do it, since it's setting the option on a transaction created in a function not used as part of this loop. That function is called at line 398 below.

The one I expect to set raw access is [here](https://github.com/apple/foundationdb/blob/01654488ba9905433bc44bd24d3082d6ca0e5978/fdbserver/workloads/FuzzApiCorrectness.actor.cpp#L349), but it is guarded by the `useSystemKeys` boolean.

Reply:
This transaction doesn't look to be reading anything; maybe this suggests that a raw transaction is being allowed to commit in `required` mode without enabling raw access, which would be an unrelated bug.

Reply:
We suspect that there may be a problem here that is orthogonal to this PR, so we don't need to block merging on it.